### DISCUSSION TITLE
Azure Function application is unable to find the certificates from keyvault.

### DISCUSSION THREAD

**Discussion by memalli03** [OrchardCMS/OrchardCore#11801] [Q&A]
Azure application is unable to find the certificates from keyvault. Proper access policy has been provided in the keyvault and even the certs are imported to azure function in TSL as private certs and even the WEBSITE_LOAD_CERTIFICATES parameter is added and all the thumbprints have been added comma separated. 

Mainly particular certificate wrt S2S is not getting identified/imported. Am I missing any configurations wrt to anything?
@kevinchalet 
Thanks

**Comment by memalli03** [comment]
@kevinchalet Can you provide some insights to this ? Or direct me to the right person who can help on this ?

**Reply by Skrypt** [reply]
I think you can reach @kevinchalet on Gitter on the OpenIddict channel. Also, don't forget to give him some donations.

**Reply by kevinchalet** [reply]
Unfortunately, I'm not really familiar with how Azure Functions work with certificates. It's not even clear to me whether you can host OC on Azure Functions: are you still using the in-process model?

**Comment by memalli03** [comment]
Update on this. I checked the azure machine file structure and the local cert store of the machine did not have the certificates so that means its not able to load it from the keyvault. But the system id policy access has been added to the necessary keyvault and they have been imported using TSA also but unfortunately not able to identify them when the app starts. 

So either the certs were never imported to the local system or there was a (Keyvault) access issue but there is no error as such

**Reply by jtkech** [reply]
Just to know, are you using `AddOrchardCoreAzureKeyVault()` ?

**Comment by memalli03** [comment]
No. I don't think so

**Reply by jtkech** [reply]
It uses a custom `AzureKeyVaultSecretManager`

    public class AzureKeyVaultSecretManager : KeyVaultSecretManager
    {
        public override string GetKey(KeyVaultSecret secret) =>
            secret.Name.Replace("---", "_").Replace("--", ":");
    }

**Reply by jtkech** [reply]
Maybe take a look at #6422 and #6359
